content/number-theory/factor.h

```diff
@@ -1,52 +1,35 @@
 /**
- * Author: Lukas Polacek
- * Date: 2010-01-28
+ * Author: chilli, SJTU
+ * Date: 2019-04-24
  * License: CC0
- * Source: Wikipedia
+ * Source: https://github.com/FTRobbin/Dreadnought-Standard-Code-Library
  * Description: Pollard's rho algorithm. It is a probabilistic factorisation
- * algorithm, whose expected time complexity is good. Before you start using it,
- * run {\tt init(bits)}, where bits is the length of the numbers you use.
- * Returns factors of the input without duplicates.
+ * algorithm, whose expected time complexity is good.
  * Time: Expected running time should be good enough for 50-bit numbers.
```

Comment:
`O(n^{1/4})` expected time, IIRC? "good enough for 50-bit numbers" is terribly vague, and probably out of date.

Reply:
It's `O(sqrt(lowest prime factor of composite))`, so `O(N^{1/4})` worst case.

Perhaps it'd be good to state that it's much faster than that on most numbers though.
I'd like to say something like
`On worst case numbers,  `O(N^{1/4})` expected runtime.`
or maybe we can just write the above definition.